src/lightmap.cpp

```diff
@@ -1249,46 +1250,124 @@ void map::apply_light_arc( const tripoint &p, const units::angle &angle, float l
 
     apply_light_source( p, LIGHT_SOURCE_LOCAL );
 
+    const point p2( p.xy() );
+
+    auto &cache = get_cache( p.z );
+    four_quadrants( &lm )[MAPSIZE_X][MAPSIZE_Y] = cache.lm;
+    float( &transparency_cache )[MAPSIZE_X][MAPSIZE_Y] = cache.transparency_cache;
+
     // Normalize (should work with negative values too)
-    const units::angle wangle = wideangle / 2.0;
+    units::angle wangle = wideangle / 2.0;
 
-    units::angle nangle = fmod( angle, 360_degrees );
+    units::angle nangle = normalize( angle );
+    units::angle oangle = angle - wangle;
+    units::angle cangle = angle + wangle;
 
     tripoint end;
     int range = LIGHT_RANGE( luminance );
     calc_ray_end( nangle, range, p, end );
-    apply_light_ray( lit, p, end, luminance );
 
     tripoint test;
     calc_ray_end( wangle + nangle, range, p, test );
 
     const float wdist = hypot( end.x - test.x, end.y - test.y );
```

Comment:
BTW although not involved in this pull request, `wdist` should be declared `double` to save single/double conversions
```suggestion
    const double wdist = hypot( end.x - test.x, end.y - test.y );
```
currently the return value from `hypot` is converted to single precision and then compared to `0.5`:
```assembly
    0x103bc0470 <+380>:  bl     0x103bc97a4               ; hypot<int, int> at math.h:1267
    0x103bc0474 <+384>:  fcvt   s0, d0
    0x103bc0478 <+388>:  fmov   s1, #0.50000000
    0x103bc047c <+392>:  fcmp   s0, s1
```